Extension/src/LanguageServer/client.ts

```diff
@@ -1486,34 +1483,31 @@ export class DefaultClient implements Client {
                     ask.Value = true;
                 }
                 if (ask.Value) {
-                    ui.showConfigureCustomProviderMessage(() => {
+                    ui.showConfigureCustomProviderMessage(async () => {
                         const message: string = (vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 1)
                             ? localize("provider.configure.folder", "{0} would like to configure IntelliSense for the '{1}' folder.", provider.name, this.Name)
                             : localize("provider.configure.this.folder", "{0} would like to configure IntelliSense for this folder.", provider.name);
                         const allow: string = localize("allow.button", "Allow");
                         const dontAllow: string = localize("dont.allow.button", "Don't Allow");
                         const askLater: string = localize("ask.me.later.button", "Ask Me Later");
-
-                        return vscode.window.showInformationMessage(message, allow, dontAllow, askLater).then(result => {
-                            switch (result) {
-                                case allow: {
-                                    this.configuration.updateCustomConfigurationProvider(provider.extensionId).then(() => {
-                                        onRegistered();
-                                        ask.Value = false;
-                                        telemetry.logLanguageServerEvent("customConfigurationProvider", { "providerId": provider.extensionId });
-                                    });
-                                    return true;
-                                }
-                                case dontAllow: {
-                                    ask.Value = false;
-                                    break;
-                                }
-                                default: {
-                                    break;
-                                }
+                        const result: string | undefined = await vscode.window.showInformationMessage(message, allow, dontAllow, askLater);
+                        switch (result) {
```

Comment:
I'd prefer `switch (await vscode.window.showInformationMessage(message, allow, dontAllow, askLater))`.

There's are a 3 more similar cases below where the await result is only used once in a switch.

Reply:
it turned out that we can't await on `showInformationMessage`, so I reverted this part of code and similar cases.

Reply:
Why can't we await this?  It seems to compile for me.  (and I actually preferred the original approach with the local variable defined before the switch, but either way is fine :))